commands/system/help.js

```diff
@@ -9,75 +10,146 @@ class Help extends Command {
       description: "Get help on a command, command category, or a setting",
       extended: "This command will display all available commands for your permission level, with the additonal option of getting per command information when you run 'help <command name>'.",
       category: "System",
-      usage: "help <category/command/setting> [page-num]",
+      usage: "help <category/command/setting>",
       aliases: ["h", "halp", "commands"]
     });
   }
 
-  async run(message, [type, page], level) {
+  async pages(message, helpmessage, pagenumber, sorted, type, level, reactions, direction) {
+    let num = 0;
+    let output = "";
+    const pg = Number(pagenumber);
+    for (const cmd of sorted.values()) {
+      if (cmd.help.category.toLowerCase() === type) {
+        if (cmd.help.category === "NSFW" && !message.channel.nsfw) continue;
+        if (num < perpage * pg && num > perpage * pg - (perpage + 1)) {
+          if (level < this.client.levelCache[cmd.conf.permLevel]) continue;
+          output += `\n\`${message.settings.prefix + cmd.help.name}\` | ${cmd.help.description.length > 80 ? `${cmd.help.description.slice(0, 80)}...` : cmd.help.description}`;
```

Comment:
Am I right in saying that you reconstruct the embed for each page, every time someone cycles forward/backward?

Reply:
(why did this show up as a review >.<)